### PR TITLE
Specify module tsc option

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- Set `compilerOptions.module` to `"Node16"`

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — APPROVED** [review]

File: docs/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "extends": "@tsconfig/docusaurus/tsconfig.json",
```
Comment (ansermino) [inline docs/tsconfig.json:3]:
```suggestion
  "extends": "@docusaurus/tsconfig",
```

You will need to install this: `yarn add --dev `@docusaurus/tsconfig`

Comment (ansermino) [reply docs/tsconfig.json:3]: See https://github.com/facebook/docusaurus/pull/9050

**Review by ansermino — CHANGES_REQUESTED** [review]
I'm guessing it was caused by this change: https://github.com/ChainSafe/forest/pull/5091

It seems the scope of this docs check is insufficient. Please add `/docs` path to the workflow, or I can do it in a subsequent PR.

File: docs/tsconfig.json (hunk 1 above)
Comment (elmattic) [reply docs/tsconfig.json]: Hmmm, the lock file looks quite different now. Could that be a problem?

Comment (ansermino) [reply docs/tsconfig.json]: That is expected, should be an issue

**Comment by ansermino** [comment]
@elmattic FYI - https://github.com/ChainSafe/forest/pull/5102

**Review by ansermino — APPROVED** [review]

**Pull request merged**